src/NHibernate.Test/NHSpecificTest/GH3325/Fixture.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH3325
+{
+	[TestFixture]
+	public class Fixture : BugTestCase
+	{
+		protected override void OnTearDown()
+		{
+			using (var session = OpenSession())
+			using (var transaction = session.BeginTransaction())
+			{
+				session.CreateQuery("delete from ChildEntity").ExecuteUpdate();
+				session.CreateQuery("delete from System.Object").ExecuteUpdate();
+
+				transaction.Commit();
+			}
+		}
+
+		[Test]
+		public void CanAddChildAfterFlush()
+		{
+			Guid parentId;
+			Guid childId;
+			using (var session = OpenSession())
+			using (var t = session.BeginTransaction())
+			{
+				var parent = new Entity { Name = "Parent" };
+				var child = new ChildEntity { Name = "Child" };
+				var parentChildren = parent.Children;
+				parentChildren.Add(child);
```

Comment:
Not sure why there could be differences in behaviour? Could you explain please @bahusoid?

Reply:
Ok, now I see..

However the test is incorrect in itself. I'll adjust the tests.